NuRadioMC/EvtGen/generator_skeleton.py

```diff
@@ -30,37 +31,72 @@ def generate_my_events(filename, n_events):
+    # In principle only the shower direction (zeniths, azimuths fields), the shower position (xx, yy, zz fields)
+    # the shower energy, the shower type and the event group id are required. We set them first
     data_sets = {}
+    # the direction of the shower
     data_sets["azimuths"] = np.ones(n_events)
     data_sets["zeniths"] = np.ones(n_events)
+    # the position of the shower
     data_sets["xx"] = np.ones(n_events)
     data_sets["yy"] = np.ones(n_events)
     data_sets["zz"] = np.ones(n_events)
-    data_sets["event_ids"] = np.arange(n_events)
+    # the shower energy
+    data_sets["shower_energies"] = np.ones(n_events) * 1 * units.EeV
+    # the shower type (here we only generate hadronic showers). This infomration is needed for the Askaryan emission model
+    data_sets["shower_type"] = ['had'] * n_events
+    # give each shower a unique id (we can also have multiple showers for a single event by just giving several showers
+    # the same event_group_id)
+    data_sets["event_group_ids"] = np.arange(n_events)
+
+    # there are a couple of additional parameters that are required to run a NuRadioMC simulations. These parameters
+    # don't influence the simulated radio signals but are required for other post analysis tasks. If these parameters
+    # are not relevant for the type of data you're generating, just set them to any value.
+
+    # specify which interaction it is (only relevant if multiple showers from the same initial neutrino are simulated)
+    # here it is just 1 for all events.
     data_sets["n_interaction"] = np.ones(n_events, dtype=np.int)
 
-    # generate neutrino flavors randomly
-    data_sets["flavors"] = np.array([12 for i in range(n_events)])
+    # the neutrino flavor. Here we only generate electron neutinos which have the integer code 12.
+    # the neutrino flavor is only used in the calculation of the "weight", i.e. the probability of the neutrino reaching
+    # the detector. If other particles than a neutrino are simulated, just set the flavor to the corresponding particle code
+    # following https://pdg.lbl.gov/2019/reviews/rpp2019-rev-monte-carlo-numbering.pdf or just set it to zero.
+    data_sets["flavors"] = 12 * np.ones(n_events, dtype=np.int)
+    # the neutrino energy. This field is also only used for the weight calculation.
     data_sets["energies"] = np.ones(n_events) * 1 * units.EeV
 
-    data_sets["interaction_type"] = np.ones(n_events, dtype='S2')
-    data_sets["inelasticity"] = np.ones(n_events) *  0.5
+    # optionally one can also directly set the event weight here (useful if particles other than neutrinos, or calibration
+    # setups are simulated
+    # data_sets["weights"] = np.ones(n_events)
 
+    # the interaction type. For neutrino interactions is can be either CC or NC. This parameter is not used but passed
+    # to the output file for information purposes.
+    data_sets["interaction_type"] = np.ones(n_events, dtype='S2')
```

Comment:
generally for the skeleton, np.ones/np.zeros are perfectly fine. For the 2 letter strings, maybe it is more intuitive to fill it with one of the actual values?

i.e. `np.ones(n_events, dtype='S2')` -> `np.full(n_events, 'CC', dtype='S2')`

Reply:
good point, will change that.

Reply:
@cg-laser another comment:

I guess fixing this resolves the future unimplemented warning when generating a terminal dump of the file:

/Users/shallmann/miniconda/lib/python3.8/site-packages/tables/attributeset.py:308: DataTypeWarning: Unsupported type for attribute 'deposited' in node '/'. Offending HDF5 class: 8
  value = self._g_getattr(self._v_node, name)
/ (RootGroup) ''
/Users/shallmann/miniconda/lib/python3.8/site-packages/tables/group.py:1201: UserWarning: problems loading leaf ``/interaction_type``::

  variable length strings are not supported yet

The leaf will become an ``UnImplemented`` node.
  warnings.warn(
/Users/shallmann/miniconda/lib/python3.8/site-packages/tables/group.py:1201: UserWarning: problems loading leaf ``/shower_type``::

  variable length strings are not supported yet

The leaf will become an ``UnImplemented`` node.

Reply:
but do we actually only ever store CC or NC there? If I remember correctly there was also a "tau_had", "tau_em" at some place stored? Or was that in another field?

Reply:
at the moment we don't use it, but we might in the future. The thing is, this field is not used at all in the simulation but you still want to know it for post analyses. In the secondary shower generation, one could use this field to save what type of interaction happend (and I think this is already being done). What matters for the simulation is the "shower_type" which is so far only "em" or "had". But in principle there could be more channels, e.g. a tau decay is slightly differnt.

Reply:
ok, so at least for the shower type we should make sure the string length is long enough, like `dtype='S7'`?